GitUI/Infrastructure/PuttyHelpers.cs

```diff
@@ -0,0 +1,102 @@
+        public static void StartPageant(string workingDirectory)
+        {
+            // Exceptions must be thrown from the UI thread, otherwise we'll crash the app.
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            ThrowIfFileNotFound(AppSettings.Pageant, $"'{AppSettings.Pageant}'\r\n\r\n{TranslatedStrings.ErrorSshPuTTYInstalled}");
+
+            new Executable(AppSettings.Pageant, workingDirectory).Start();
+        }
+
+        /// <summary>
+        ///  Starts PuTTY key generator for the specified working directory.
+        /// </summary>
+        /// <param name="workingDirectory">The working directory.</param>
+        /// <exception cref="UserExternalOperationException">PuTTY executable cannot be found.</exception>
+        /// <exception cref="COMException">The method was not called on the UI thread.</exception>
+        public static void StartPuttygen(string workingDirectory)
+        {
+            // Exceptions must be thrown from the UI thread, otherwise we'll crash the app.
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            ThrowIfFileNotFound(AppSettings.Puttygen, $"'{AppSettings.Puttygen}'\r\n\r\n{TranslatedStrings.ErrorSshPuTTYInstalled}");
+
+            new Executable(AppSettings.Puttygen, workingDirectory).Start();
+        }
+
+        /// <summary>
+        ///  Starts PuTTY agent with the specified SSH key.
+        /// </summary>
+        /// <param name="sshKeyFileLoader">The delegate that provides the key to load.</param>
+        /// <returns><see langword="true"/> if PuTTY agent was started successfully; otherwise, <see langword="false"/>.</returns>
+        /// <exception cref="UserExternalOperationException">
+        ///  <para>PuTTY isn't configured is preferred SSH client.</para>
+        ///  <para>- or -</para>
+        ///  <para>PuTTY executable cannot be found.</para>
+        ///  <para>- or -</para>
+        ///  <para>The SSH key cannot be found.</para>
+        /// </exception>
+        /// <exception cref="COMException">The method was not called on the UI thread.</exception>
+        public static bool StartPageantIfConfigured(Func<string?> sshKeyFileLoader)
+        {
+            // Exceptions must be thrown from the UI thread, otherwise we'll crash the app.
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (!GitSshHelpers.IsPlink)
+            {
+                throw new UserExternalOperationException(TranslatedStrings.ErrorSshPuTTYWhereConfigure,
+                    new ExternalOperationException(innerException: new Exception(TranslatedStrings.ErrorSshPuTTYNotConfigured)));
+            }
+
+            ThrowIfFileNotFound(AppSettings.Pageant, $"'{AppSettings.Pageant}'\r\n\r\n{TranslatedStrings.ErrorSshPuTTYInstalled}");
+
+            string? sshKeyFile = sshKeyFileLoader();
+            ThrowIfFileNotFound(sshKeyFile, $"'{sshKeyFile}'", TranslatedStrings.ErrorSshKeyNotFound);
```

Comment:
https://github.com/gitextensions/gitextensions/issues/10530

Reply:
Yes, thanks. It looks like it has to be rolled back.